flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeSchedulerNG.java

```diff
@@ -640,7 +645,11 @@ private void runIfState(State expectedState, Runnable action, long delay, TimeUn
         componentMainThreadExecutor.schedule(
                 () -> {
                     if (state == expectedState) {
-                        action.run();
+                        try {
+                            action.run();
+                        } catch (Throwable t) {
+                            LOG.warn("Error in runnable", t);
```

Comment:
Chances are that this should go straight to the FatalExitExceptionHandler#INSTANCE.